documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc

```diff
@@ -891,6 +891,27 @@ refer to the documentation:
 {elasticsearchDocUrl}/analysis-tokenfilters.html[token filters].
 ====
 
+== Other index settings
+
+=== Max result window size
+
+It is possible to set the maximum window size for a query result.
+
+[source]
+----
+# To configure the defaults for all indexes:
+hibernate.search.backend.searching.max_result_window_size = 5000
+# To configure a specific index:
+hibernate.search.backend.indexes.<index name>.searching.max_result_window_size = 5000
+----
+The default for this property is 10000.
+
+[NOTE]
+====
+This value will be used to limit the returning hits size if no limit has been defined on the query by the user.
+In this case if Hibernate Search notices there are more results, a warning will be issued.
+====
+
 == Custom index settings
```

Comment:
What happens if someone sets the max result window size through custom index settings? Do we take that into account? Is there a test for that?

Reply:
Actually... Shouldn't we rely exclusively on custom index settings, and remove the dedicated property? I think it would make sense, don't you?